packages/integration_test/lib/common.dart

```diff
@@ -297,6 +297,9 @@ abstract class CallbackManager {
   /// Android only. Converts the Flutter surface to an image view.
   Future<void> convertFlutterSurfaceToImage();
 
+  /// Android only. Converts the image view to the Flutter surface.
+  Future<void> revertFlutterImage();
```

Comment:
> What is the use case for toggling between an Image backed surface and a non-Image backed surface in a test? Flutter developers cannot do this from their apps.

Solely for the purpose for taking screenshots... I see the point here.

From the app developer's perspective, I'm not interested in managing the Flutter surface. 

I wonder if we can call `takeScreenshot` without having to call `convertFlutterSurfaceToImage` and `revertFlutterImage`. 

This is in `IOCallbackManager`

```dart
  @override
  Future<Map<String, dynamic>> takeScreenshot(String screenshot, [Map<String, Object?>? args]) async {
    assert(args == null, '[args] handling has not been implemented for this platform');
    if (Platform.isAndroid && !_isSurfaceRendered) {
     // Why don't we just call `convertFlutterSurfaceToImage` here?
      throw StateError('Call convertFlutterSurfaceToImage() before taking a screenshot');
    }
    integrationTestChannel.setMethodCallHandler(_onMethodChannelCall);
    final List<int>? rawBytes = await integrationTestChannel.invokeMethod<List<int>>(
      'captureScreenshot',
      <String, dynamic>{'name': screenshot},
    );
   // Why don't we revert it here? 
    if (rawBytes == null) {
      throw StateError('Expected a list of bytes, but instead captureScreenshot returned null');
    }
    return <String, dynamic>{
      'screenshotName': screenshot,
      'bytes': rawBytes,
    };
  }
```

Reply:
I think this was the suggestion of @johnmccutchan but there is one elementary problem. Converting the surface does not happen immediately in tests. As far as i know, you have to call .pumpAndSettle() on your WidgetTester instance which is not available in the binding context.
That is, what the documentation of IntegrationTestWidgetsFlutterBinding.takeScreenshot() says: 
```doc
Takes a screenshot.
On Android, you need to call convertFlutterSurfaceToImage(), and pump a frame before taking a screenshot.
```